catalog/mgmp/catalog-transformer-mgmp/src/main/java/org/codice/alliance/catalog/mgmp/transformer/MgmpTransformer.java

```diff
@@ -0,0 +1,594 @@
+                false);
+        addValuesFromPathToListIfPresent(pathValueTracker,
+                mgmpTypes,
+                MgmpConstants.MGMP_SPATIAL_REFERENCE_SYSTEM_GRID_NUMBER_TYPE_PATH,
+                false);
+
+        ArrayList<String> mgmpCodes = new ArrayList<>();
+        addValuesFromPathToListIfPresent(pathValueTracker,
+                mgmpCodes,
+                MgmpConstants.MGMP_SPATIAL_REFERENCE_SYSTEM_TYPE_PATH,
+                false);
+        addValuesFromPathToListIfPresent(pathValueTracker,
+                mgmpCodes,
+                MgmpConstants.MGMP_SPATIAL_REFERENCE_SYSTEM_DATUM_TYPE_PATH,
+                false);
+        addValuesFromPathToListIfPresent(pathValueTracker,
+                mgmpCodes,
+                MgmpConstants.MGMP_SPATIAL_REFERENCE_SYSTEM_ELLIPSOID_TYPE_PATH,
+                false);
+        addValuesFromPathToListIfPresent(pathValueTracker,
+                mgmpCodes,
+                MgmpConstants.MGMP_SPATIAL_REFERENCE_SYSTEM_PROJECTION_TYPE_PATH,
+                false);
+        addValuesFromPathToListIfPresent(pathValueTracker,
+                mgmpCodes,
+                MgmpConstants.MGMP_SPATIAL_REFERENCE_SYSTEM_GRID_TYPE_PATH,
+                false);
+
+        List<String> crsCodes = new ArrayList<>();
+        addValuesFromPathToListIfPresent(pathValueTracker,
+                crsCodes,
+                MgmpConstants.MGMP_SPATIAL_REFERENCE_SYSTEM_CRS_WKT_TYPE_PATH,
+                true);
+        addValuesFromPathToListIfPresent(pathValueTracker,
+                crsCodes,
+                MgmpConstants.MGMP_SPATIAL_REFERENCE_SYSTEM_GRID_WKT_TYPE_PATH,
+                true);
+        addValuesFromPathToListIfPresent(pathValueTracker,
+                crsCodes,
+                MgmpConstants.MGMP_SPATIAL_REFERENCE_SYSTEM_PROJECTION_WKT_TYPE_PATH,
+                true);
+        addValuesFromPathToListIfPresent(pathValueTracker,
+                crsCodes,
+                MgmpConstants.MGMP_SPATIAL_REFERENCE_SYSTEM_ELLIPSOID_WKT_TYPE_PATH,
+                true);
+        addValuesFromPathToListIfPresent(pathValueTracker,
+                crsCodes,
+                MgmpConstants.MGMP_SPATIAL_REFERENCE_SYSTEM_DATUM_WKT_TYPE_PATH,
+                true);
+
+        List<String> types = pathValueTracker.getAllValues(toPath(GmdConstants.CRS_CODE_PATH));
+        List<String> codes = pathValueTracker.getAllValues(toPath(GmdConstants.CRS_AUTHORITY_PATH));
+        if (CollectionUtils.isNotEmpty(codes) && CollectionUtils.isNotEmpty(types)) {
+            for (int i = 0; i < codes.size(); i++) {
+                crsCodes.add(codes.get(i) + ":" + types.get(i));
+            }
+        }
+
+        if (CollectionUtils.isNotEmpty(mgmpTypes) && CollectionUtils.isNotEmpty(mgmpCodes)) {
+            for (int i = 0; i < mgmpCodes.size(); i++) {
+                crsCodes.add(mgmpTypes.get(i) + ":" + mgmpCodes.get(i));
```

Comment:
On line 264, you are doing CODES:TYPES, and here you are doing TYPES:CODES. Seems confusing.